okteto/okteto
# [Hybrid] Warn on unused fields but don't fail

# Proposed changes

Fixes https://github.com/okteto/okteto/issues/3685

## Description

This change adds warnings that will be displayed to the users when they enable hybrid mode in a manifest where the `dev` section has fields unused by hybrid mode, but that could belong to the default mode. This can occur when trying hybrid mode on an existing configured manifest.

## How to test the old behavioir

1. Clone a sample repo: `git clone git@github.com:andreafalzetti/hybrid-test.git`
2. Go inside and using the latest okteto CLI version, run: `cd hybrid-test && okteto up`
3. Notice a series of errors and the process interrupted

## How to test the new behaviour

1. Checkout this branch and build the CLI
2. Go to the previous sample repo and run `okteto up` again.
3. Notice that the command is not failing anymore and you should see a series of warn messages saying: "In hybrid mode, the field "<field>" specified in your manifest is ignored".
